### PR TITLE
NodeJS: Switch to use new Fathead Template Elements

### PR DESCRIPTION
**Pull request by bfmags**
## NodeJS ##

- [x] implement #412 -- Switch to use new Fathead Template Elements

___________________
IA Page:  http://duck.co/ia/view/node_js

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/nodejs_ref/parse.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ function getAbstract(obj) {
         stability = '<p>Stability: ' + obj.stability + ' - ' + obj.stabilityText + '</p>';
     }
 
-    a = code + a + stability;
+    a = '<section class="prog__container">' a + code + stability + '</section>';
```
Comment (moollaza) [inline lib/fathead/nodejs_ref/parse.js]: looks like you have a typo here? Should it be `'<section class="prog__container">' + a + code` ?

After changing that I'm also getting another error:

```
/mnt/ebs/usr/local/ddh/zeroclickinfo-fathead/lib/fathead/nodejs_ref/parse.js:108
    json[sectionKey].forEach(function(node) {
                    ^

TypeError: Cannot read property 'forEach' of undefined
    at /mnt/ebs/usr/local/ddh/zeroclickinfo-fathead/lib/fathead/nodejs_ref/parse.js:108:21
    at Array.forEach (native)
    at Object.<anonymous> (/mnt/ebs/usr/local/ddh/zeroclickinfo-fathead/lib/fathead/nodejs_ref/parse.js:107:13)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:974:3
```

Comment (bfmags) [inline lib/fathead/nodejs_ref/parse.js]: thnx moollaza
nodejs api reference file all.json as been updated ('methods' key no longer present)
need to update the code to reflect the changes

**Comment by bfmags** [comment]
fixed typo
tested code OK with a previous version of nodejs API reference

Parser needs to be updated to latest API reference
https://github.com/duckduckgo/zeroclickinfo-fathead/issues/436

**Comment by moollaza** [comment]
@bfmags there's one more small bug -- the dir for this needs to be identical to the ID of the IA. This should have been changed a while ago, but I guess I hadn't noticed.

The dir for this fathead should be `/lib/fathead/node_js`

Can you please update that as well? After that this should be good for merge 👍

**Comment by bfmags** [comment]
@moollaza should be good to go

**Pull request merged**